### PR TITLE
Add subsection headers for the mangling documentation.

### PR DESCRIPTION
**Pull request by sunfishcode**
This enables linking to specific subsections.

### CODE REVIEW

File: design/mvp/CanonicalABI.md [hunk 1]
```diff
@@ -1429,6 +1429,8 @@ character in a component-level import/export (as is currently the case in `wit`
 [identifiers](WIT.md#identifiers)) and thus can safely be used to prefix
 auxiliary Canonical ABI-induced imports/exports.
 
+#### Instance mangling
```
Comment (lukewagner) [inline design/mvp/CanonicalABI.md:1432]:
```suggestion
#### Instance type mangling
```

File: design/mvp/CanonicalABI.md [hunk 2]
```diff
@@ -1429,6 +1429,8 @@ character in a component-level import/export (as is currently the case in `wit`
...
+
 Instance-mangling recursively builds a dotted path string (of instance names)
```
Comment (lukewagner) [inline design/mvp/CanonicalABI.md:1434]:
```suggestion
Instance-type mangling recursively builds a dotted path string (of instance names)
```

File: design/mvp/CanonicalABI.md [hunk 3]
```diff
@@ -1457,6 +1459,8 @@ def mangle_instances(xs, path = ''):
 The three `TODO` cases are intended to be filled in by future PRs extending
 the Canonical ABI.
 
+#### Function and value type mangling
```
Comment (lukewagner) [inline design/mvp/CanonicalABI.md:1462]:
```suggestion
#### Function type mangling
```

Comment (lukewagner) [reply design/mvp/CanonicalABI.md:1462]: And perhaps add a "Value type mangling" below?

**Review by lukewagner — APPROVED** [review]
Maybe have separate "Function type mangling" and "Value type

File: design/mvp/CanonicalABI.md (hunk 3 above)
Comment (sunfishcode) [reply design/mvp/CanonicalABI.md]: Done

**Comment by lukewagner** [comment]
Thanks!

**Pull request merged**